### PR TITLE
Implement Filter Bar with all filterls (List Report)

### PR DESCRIPTION
**Pull request by BabichevDima**

### CODE REVIEW

File: project007/webapp/controller/ListReport.controller.js [hunk 1]
```diff
@@ -271,6 +242,42 @@ sap.ui.define(
...
+        var aFilters            = [];
+
+        if (sAllFieldValue && !isNaN(sAllFieldValue)){
+          aFilters.push(new Filter("ID", FilterOperator.EQ, sAllFieldValue));
+        } else if (sAllFieldValue){
+          aFilters.push(new Filter("Name", FilterOperator.Contains, sAllFieldValue));
+        }
+
+        if (aCategoryNameValue) {
+          aFilters.push(...aCategoryNameValue.map((sValue) => {return new Filter("Name", FilterOperator.Contains, sValue)}));
```
Comment (koryakovskaya) [inline project007/webapp/controller/ListReport.controller.js]: ...map(sValue=>new Filter("Name", FilterOperator.Contains, sValue))

File: project007/webapp/view/ListReport.view.xml [hunk 2]
```diff
@@ -26,23 +26,16 @@
                         <fb:filterItems>
                             <fb:FilterItem name="ProductName" label="Search">
                                 <fb:control>
-                                    <Input id="searchFieldName" placeholder="{i18n>PlaceholderSerch}" valueLiveUpdate="true" liveChange="combineFilters('Name')" value="{oFilterModel>/name}"/>
+                                    <SearchField id="searchFieldName" search="combineFilters" placeholder="{i18n>PlaceholderSerch}" showRefreshButton='true' value="{stateModel>/AllField}"/>
```
Comment (koryakovskaya) [inline project007/webapp/view/ListReport.view.xml]: rename search

File: project007/webapp/controller/ListReport.controller.js [hunk 3]
```diff
@@ -271,6 +242,42 @@ sap.ui.define(
...
+        }
+        oItemsBinding.filter(aFilters)
+      },
+
+      /**
+       * Get an array of suppliers.
+       * 
+       * @returns {Array} Suppliers.
+       * 
+       * @private
```
Comment (koryakovskaya) [inline project007/webapp/controller/ListReport.controller.js:273]: add privacy in all functions:)

Comment (BabichevDima) [reply project007/webapp/controller/ListReport.controller.js:273]: @koryakovskaya I don't have enough salary=) For all private functions)

**Comment by koryakovskaya** [comment]
Approved

**Pull request merged**